File: db/event_rating_public.php
<?php

require_once("entity.php");

/**
 * Class for accessing, manipulating, creating and deleting records in event_rating_public.
*/

class Event_Rating_Public extends Entity
{

  public function __construct($select = array())
  {
    $this->table = "event_rating_public";
    $this->order = "";
    $this->domain = "public";
    $this->field['event_id']['type'] = 'INTEGER';
    $this->field['event_id']['not_null'] = true;
    $this->field['participant_knowledge']['type'] = 'SMALLINT';
    $this->field['topic_importance']['type'] = 'SMALLINT';
    $this->field['content_quality']['type'] = 'SMALLINT';
    $this->field['presentation_quality']['type'] = 'SMALLINT';
    $this->field['audience_involvement']['type'] = 'SMALLINT';
    $this->field['remark']['type'] = 'TEXT';
    $this->field['eval_time']['type'] = 'TIMESTAMP';
    $this->field['eval_time']['not_null'] = true;
    $this->field['eval_time']['default'] = true;
    $this->field['rater_ip']['type'] = 'INET';
    $this->field['event_id']['primary_key'] = true;
    $this->field['eval_time']['primary_key'] = true;
    parent::__construct($select);
  }

}

?>

File: pentabarf/lib/momomoto/tables/event_rating_public.rb
module Momomoto
  class Event_rating_public < Base
    def initialize
      super
      @domain = 'public'
      @fields = {
        :event_id => Datatype::Integer.new( {:not_null=>true, :primary_key=>true} ),
        :participant_knowledge => Datatype::Smallint.new( {} ),
        :topic_importance => Datatype::Smallint.new( {} ),
        :content_quality => Datatype::Smallint.new( {} ),
        :presentation_quality => Datatype::Smallint.new( {} ),
        :audience_involvement => Datatype::Smallint.new( {} ),
        :remark => Datatype::Text.new( {} ),
        :eval_time => Datatype::Timestamp.new( {:with_timezone=>true, :not_null=>true, :default=>true, :primary_key=>true} ),
        :rater_ip => Datatype::Inet.new( {} )
      }
    end
  end
end
